src/parser.rs

```diff
@@ -4180,6 +4180,25 @@ impl<'a> Parser<'a> {
             SetExpr::Query(Box::new(subquery))
         } else if self.parse_keyword(Keyword::VALUES) {
             SetExpr::Values(self.parse_values()?)
+        } else if self.parse_keyword(Keyword::TABLE) {
+            self.next_token();
+            self.next_token();
+            self.next_token();
```

Comment:
Rather than consuming these tokens and the starting all the way back at `tokens[0]` the typical pattern in this codebase would be to call `self.parse_as_table` which would assume the token stream was already at `CREATE TABLE name AS ...` and only parse the name.

I wonder if you could use `parse_identifier` too rather than parsing schema / table name directly